code/modules/tools/tool_extension.dm

```diff
@@ -0,0 +1,39 @@
+/datum/extension/tool
+	expected_type = /obj/item
+	base_type = /datum/extension/tool
+	var/list/tool_values     // Delay multipliers/general tool quality indicators, lower is better but 0 or lower means it isn't valid as a tool.
+	var/list/tool_use_sounds // Associative list of tool to sound/list of sounds used to override the archetype config sounds.
+
+/datum/extension/tool/New(datum/holder, list/_tool_values, list/_tool_use_sounds)
+	..()
+	tool_values = _tool_values
+	tool_use_sounds = _tool_use_sounds
```

Comment:
I think you should just set these on the extension after creating it (get_extension(); extension.sounds = blah) for better clarity. The one place this is used there are two lists of sounds and it's really confusing to read.

Reply:
I'm not sure I agree that it's confusing to read, it's only a couple of extra lines. Setting them after the fact would require retrieving the extension, setting the vars, and then having the extension rebuild the sound lists as it currently does in Initialize - it wouldn't be any more readable than a couple of extra lines in a constructor.

Reply:
With the discovery of `get_or_create_extension` I went ahead and refactored those procs to accept shorter arguments supplied post-extension-setting.